ppa/archive/admin.py

```diff
@@ -33,16 +33,15 @@ def save_related(self, request, form, formsets, change):
 
     def bulk_add_collection(self, request, queryset):
         '''
-        Bulk add a queryset of :class:ppa.archive.DigitizedWork to
-        a :class:ppa.archive.Collection.
+        Bulk add a queryset of :class:`ppa.archive.DigitizedWork` to
+        a :class:`ppa.archive.Collection`.
         '''
         # Uses POST from admin rather than a database query to get the pks
         # per the suggested practices in Django documentation
-        selected = request.POST.getlist(admin.ACTION_CHECKBOX_NAME)
-
-        return HttpResponseRedirect(
-            '%s?ids=%s' % (reverse('archive:bulk-add'), ','.join(selected))
-        )
+        selected = ','.join(str(val) for val in
+                            queryset.order_by('id').values_list('id', flat=True))
+        request.session['selected_works'] = selected
```

Comment:
Couldn't you just pass the list and let django serialize it as json for you?

Also, a more specific session key might be a good thing here. Maybe  `collection-bulkadd-ids` or something along those lines?

Reply:
I was considering a more specific key, so I like that suggestion. 

My thought was to keep it simple with a comma delimited list, but in principle I don't see why that wouldn't work. I doubt the parsing difference would be significant at all if you think a list would be more clear.

Reply:
I think it will simplify the code, which will make it more readable. Agreed, probably similar performance.